File: incubator/net.sf.j2s.java.org.eclipse.osgi/src/org/eclipse/core/runtime/internal/adaptor/EclipseBundleInstaller.java
/*******************************************************************************
 * Copyright (c) 2004 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 * 
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
package org.eclipse.core.runtime.internal.adaptor;

import org.eclipse.osgi.framework.adaptor.core.BundleInstaller;
import org.eclipse.osgi.service.resolver.BundleDescription;
import org.osgi.framework.*;

/**
 * Internal class.
 */
//TODO: minimal implementation for now. This could be smarter
public class EclipseBundleInstaller implements BundleInstaller {
	BundleContext context;
	public EclipseBundleInstaller(BundleContext context) {
		this.context = context;
	}

	public void installBundle(BundleDescription toInstall) throws BundleException {
		context.installBundle(toInstall.getLocation());
	}

	public void uninstallBundle(BundleDescription toUninstallId) throws BundleException {
		Bundle toUninstall = context.getBundle(toUninstallId.getBundleId());
		if (toUninstall != null)
			toUninstall.uninstall();
	}

	public void updateBundle(BundleDescription toUpdateId) throws BundleException {
		Bundle toUpdate = context.getBundle(toUpdateId.getBundleId());
		if (toUpdate != null)
			toUpdate.update();
	}
}

File: incubator/net.sf.j2s.test.mauve/j2slib/org/eclipse/core/runtime/internal/adaptor/EclipseBundleInstaller.js
Clazz.declarePackage ("org.eclipse.core.runtime.internal.adaptor");
Clazz.load (["org.eclipse.osgi.framework.adaptor.core.BundleInstaller"], "org.eclipse.core.runtime.internal.adaptor.EclipseBundleInstaller", null, function () {
c$ = Clazz.decorateAsClass (function () {
this.context = null;
Clazz.instantialize (this, arguments);
}, org.eclipse.core.runtime.internal.adaptor, "EclipseBundleInstaller", null, org.eclipse.osgi.framework.adaptor.core.BundleInstaller);
Clazz.makeConstructor (c$, 
function (context) {
this.context = context;
}, "org.osgi.framework.BundleContext");
Clazz.overrideMethod (c$, "installBundle", 
function (toInstall) {
this.context.installBundle (toInstall.getLocation ());
}, "org.eclipse.osgi.service.resolver.BundleDescription");
Clazz.overrideMethod (c$, "uninstallBundle", 
function (toUninstallId) {
var toUninstall = this.context.getBundle (toUninstallId.getBundleId ());
if (toUninstall != null) toUninstall.uninstall ();
}, "org.eclipse.osgi.service.resolver.BundleDescription");
Clazz.overrideMethod (c$, "updateBundle", 
function (toUpdateId) {
var toUpdate = this.context.getBundle (toUpdateId.getBundleId ());
if (toUpdate != null) toUpdate.update ();
}, "org.eclipse.osgi.service.resolver.BundleDescription");
});
